meta/st0x-spec.md

```diff
@@ -0,0 +1,96 @@
+# st0x Server API Specification V1
+
+## Goal
+
+Provide signed price quotes for specific trading pairs that can be used by solvers in Raindex orders. Each endpoint handles a single trading pair and returns a signed context containing quote information.
+
+## Magic Numbers
+
+- Web data metadata magic number: `0xff5dcce9b571ba42`
+- st0x API magic number: `0xff5d577062616972` ("st0x pair" in hex)
+
+## Server Advertisement
+
+Servers advertise price quote endpoints on metaboard with:
+- 8 byte magic number for st0x API
+- Remaining bytes contain UTF-8 encoded URL where the API is hosted for a specific trading pair
+
+Each trading pair requires its own advertisement. Orders reference these advertisements by their hash to save gas.
+
+## Request Format
+
+Clients (solvers) POST raw octet streams to the server URL with the following binary format:
+
+```
+[ 65 byte ECDSA signature ]  # Signs the hash of all following bytes
+[ 32 byte keccak256 hash ]   # Hash of web data from metaboard
+[ 8 byte unix timestamp ]    # Seconds since epoch
+[ 32 byte nonce ]           # Random bytes for uniqueness
+```
+
+No additional payload needed since each endpoint is pair-specific.
+
+## Server Response
+
+The server returns a SignedContextV1 structure containing:
+
+```solidity
+struct SignedContextV1 {
+    address signer;           // Account that signed the context
+    uint256[] context;        // The signed data array
+    bytes signature;          // EIP-191 compliant signature
```

Comment:
it's actually ERC-1271 because it uses the OZ logic

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/cryptography/SignatureChecker.sol#L14

if it was only 191 compliant then we wouldn't need the signer in the struct at all because `ecrecover` gives back the signer from the signature

the 1271 fallback is that if the signer has nonzero code onchain it will call the signer as a smart contract and ask the signer if the signature is valid